In [3]:
import pandas as pd
import numpy as np

#### Load Tables + Basic Summary

In [4]:
train_df = pd.read_csv('data_raw/train.csv')
test_df = pd.read_csv('data_raw/test.csv')

In [5]:
len(train_df), len(test_df)

(36515, 19897)

In [9]:
display(train_df.head(3))
display(test_df.head(3))

id_code experiment  plate well  sirna
0  HEPG2-01_1_B03   HEPG2-01      1  B03    513
1  HEPG2-01_1_B04   HEPG2-01      1  B04    840
2  HEPG2-01_1_B05   HEPG2-01      1  B05   1020

id_code experiment  plate well
0  HEPG2-08_1_B03   HEPG2-08      1  B03
1  HEPG2-08_1_B04   HEPG2-08      1  B04
2  HEPG2-08_1_B05   HEPG2-08      1  B05

#### GroupBy Aggregation

In [12]:
print(list(train_df.experiment.unique()))

['HEPG2-01', 'HEPG2-02', 'HEPG2-03', 'HEPG2-04', 'HEPG2-05', 'HEPG2-06', 'HEPG2-07', 'HUVEC-01', 'HUVEC-02', 'HUVEC-03', 'HUVEC-04', 'HUVEC-05', 'HUVEC-06', 'HUVEC-07', 'HUVEC-08', 'HUVEC-09', 'HUVEC-10', 'HUVEC-11', 'HUVEC-12', 'HUVEC-13', 'HUVEC-14', 'HUVEC-15', 'HUVEC-16', 'RPE-01', 'RPE-02', 'RPE-03', 'RPE-04', 'RPE-05', 'RPE-06', 'RPE-07', 'U2OS-01', 'U2OS-02', 'U2OS-03']


In [5]:
for field in (col for col in train_df.columns if col != 'id_code'):

    field_count = train_df.groupby([field]).count()['id_code']
    print(f'num of {field}: {len(field_count)}')
    print(f'min: {min(field_count)} | max: {max(field_count)}')
    display(pd.DataFrame(field_count).head(5))

num of experiment: 33
min: 1090 | max: 1108


id_code
experiment         
HEPG2-01       1106
HEPG2-02       1106
HEPG2-03       1108
HEPG2-04       1108
HEPG2-05       1108

num of plate: 4
min: 9127 | max: 9130


id_code
plate         
1         9129
2         9129
3         9130
4         9127

num of well: 277
min: 128 | max: 132


id_code
well         
B03       132
B04       132
B05       132
B06       132
B07       132

num of sirna: 1108
min: 29 | max: 33


id_code
sirna         
0           33
1           33
2           33
3           33
4           33

In [6]:
#experiments * wells = plates
33 * 277

9141

In [7]:
#wells * plates = sirna
4 * 277

1108

### Interpretation of the Experimental Scope

Of the 1108 sirna classes, there are between 29 and 33 example images or each; the same as number of experiments.

The wells are evely used (132 times each), the small variation maybe be due to train controls (?)

Key: in one well of one plate is each of the sirna for each experiment (Still speculation, but numbers match)

There only four plates? But there are only (max?) four plates per experiment...

\[x] TODO - groupby ['experiment', 'plate']

##### Misc Analysis on Train

In [13]:
# fields = ['experiment']
fields = ['experiment', 'well']
field_count = train_df.groupby(fields) #.unique()
field_count

In [9]:
?pd.DataFrame.groupby

In [10]:
?pd.DataFrame.pivot_table

In [11]:
np.count

AttributeError: module 'numpy' has no attribute 'count'

In [ ]:
train_df.pivot_table(index='sirna',columns='experiment', aggfunc=count)

### Test Data: characterize, and check for any overlap with train?

In [14]:
for field in (col for col in test_df.columns if col != 'id_code'):

    field_count = test_df.groupby([field]).count()['id_code']
    print(f'num of {field}: {len(field_count)}')
    print(f'min: {min(field_count)} | max: {max(field_count)}')
    display(pd.DataFrame(field_count).head(5))

num of experiment: 18
min: 1094 | max: 1108


id_code
experiment         
HEPG2-08       1107
HEPG2-09       1108
HEPG2-10       1108
HEPG2-11       1106
HUVEC-17       1108

num of plate: 4
min: 4971 | max: 4978


id_code
plate         
1         4978
2         4975
3         4973
4         4971

num of well: 277
min: 64 | max: 72


id_code
well         
B03        72
B04        72
B05        72
B06        72
B07        72

In [17]:
test_df.experiment.unique()

array(['HEPG2-08', 'HEPG2-09', 'HEPG2-10', 'HEPG2-11', 'HUVEC-17',
       'HUVEC-18', 'HUVEC-19', 'HUVEC-20', 'HUVEC-21', 'HUVEC-22',
       'HUVEC-23', 'HUVEC-24', 'RPE-08', 'RPE-09', 'RPE-10', 'RPE-11',
       'U2OS-04', 'U2OS-05'], dtype=object)

In [19]:
#overlap on experiments?
any([e in train_df.experiment.unique() 
     for e in test_df.experiment.unique()])

False

So: there are 33 train experiments and 18 test experiments; there is no overlap between samples.



### Explore the other data files - the Controls

Explore the controls within each experiment: are they in the mission controls?



In [21]:
import os

In [22]:
os.listdir('data_raw/')

['pixel_stats.csv',
 'sample_submission.csv',
 'test.csv',
 'test_controls.csv',
 'train.csv',
 'train_controls.csv']